docs/architecture/adr-003-remarkables-contract.md

```diff
@@ -145,18 +145,27 @@ pub struct QueryConfigResponse {
 }
 ```
 
-#### EngagementThresholds
-The `EngagementThresholds` query returns the thresholds needed to mint a particular Remarkable inside a `EngagementThresholdsResponse`.
+#### Rarities
+The `Rarities{}` query return all the info rarities required to mint a specific Remarkable inside a `QueryRaritiesResponse`.
+```rust
+pub struct QueryRaritiesResponse {
+  pub rarities: Vec<Rarity>,
+}
+```
+
+### AllNftInfo
+This query returns all the nft info of the token from cw721 contract.
 ```rust
-pub struct QueryEngagementThresholdsResponse {
-  pub engagement_thresholds: Vec<EngagementThreshold>
+pub struct AllNftInfoResponse<Metadata> {
+    pub access: OwnerOfResponse,
+    pub info: NftInfoResponse<Metadata>,
 }
 ```
 
-#### RarityMintFees
-The `RarityMintFees` query return the rarity mint fees required to mint a specific Remarkable inside a `RarityMintFeesResponse`.
+### Tokens
+This query returns all the ids of tokens owned by the given owner from cw721 contract.
 ```rust
-pub struct QueryRarityMintFeesResponse {
-  pub rarity_mint_fees: Vec<RarityMintFee>
+pub struct TokensResponse {
+    pub tokens: Vec<String>,
 }
 ```
```

Comment:
About the `extention` field, I think we can use it to add some meaningful metadata that should become useful for the use cases I envision for the remarkables. In particular, we should have such `Metadata` on chain:
```rust
pub struct Metadata {
pub rarity_level: u32,
pub subspace_id: u64,
}

```

Reply:
The level and post id is including in the token id. If the client want to know the rarity level or post id from a token, they can parse token id, besides, subspace id can be queried from the remarkables config. From my side, `Metadata` seems duplicated.

Reply:
Yeah, reading from `Metadata` field would cost less. consider the communication among each contracts, having a `Metadata` would be useful.

Reply:
Added